pkg/reconciliation/reconciliation.go

```diff
@@ -0,0 +1,53 @@
+package reconciliation
+
+import (
+	"context"
+	"time"
+
+	"github.com/k8ssandra/k8ssandra-operator/pkg/annotations"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/result"
+	errors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type k8sResource interface {
+	client.Object
+	annotations.Annotated
+	DeepCopy()
+	DeepCopyInto(out *client.Object)
+}
+
+func ReconcileObject[G k8sResource](ctx context.Context,
+	desiredObj k8sResource,
+	currentObj *k8sResource,
+	client client.Client,
+	requeueDelay time.Duration) result.ReconcileResult {
+	objectKey := types.NamespacedName{
+		Name:      desiredObj.GetName(),
+		Namespace: desiredObj.GetNamespace(),
+	}
+	annotations.AddHashAnnotation(desiredObj)
+	err := client.Get(ctx, objectKey, *currentObj)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			if err := client.Create(ctx, desiredObj); err != nil {
```

Comment:
What about setting controller references here? For example, in the Stargate controller, we have a ConfigMap [reconcile function](https://github.com/k8ssandra/k8ssandra-operator/blob/main/controllers/stargate/vector.go#L49-L51) for reconciling the Vector ConfigMap for Stargate metrics. We set the config map's controller reference to the stargate object so that cleanup happens when necessary.

Reply:
That part of the reconciliation process is discussed over [here](https://github.com/k8ssandra/k8ssandra-operator/issues/827#issuecomment-1418633875).